src/Sql/Sql/Elastic Jobs/Cmdlet/Agent/SetAzureRmSqlElasticJobAgent.cs

```diff
@@ -0,0 +1,176 @@
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlElasticJobAgent Cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlElasticJobAgent",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlElasticJobAgentModel))]
+    public class SetAzureSqlElasticJobAgent : AzureSqlElasticJobAgentCmdletBase<AzureSqlElasticJobAgentModel>
+    {
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The resource group name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent input object
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The agent input object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobAgentModel InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent resource id
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The agent resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the server name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 1,
+            HelpMessage = "The server name")]
+        [ValidateNotNullOrEmpty]
+        public override string ServerName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 2,
+            HelpMessage = "The agent name")]
+        [ValidateNotNullOrEmpty]
+        [Alias("AgentName")]
+        public override string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Agent tags
+        /// </summary>
+        [Parameter(HelpMessage = "The agent tags", ParameterSetName = DefaultParameterSet)]
+        [Parameter(HelpMessage = "The agent tags", ParameterSetName = InputObjectParameterSet)]
+        [Parameter(HelpMessage = "The agent tags", ParameterSetName = ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias("Tags")]
+        public Hashtable Tag { get; set; }
```

Comment:
@johnpaulkee are there no other properties that you can set on this type than the tag? If so, then this cmdlet shouldn't be necessary since the user can set the tags on any resource using the generic `Set-AzResource` cmdlet

Reply:
We were potentially adding updates to this cmdlet in the future to update the job vCores.
I think redundancy is ok here with Set-AzResource since it'll give us more flexibility in the future to change this if needed.